### PR TITLE
docs: mention `NuxtIsland` in server only components docs

### PR DESCRIPTION
**Pull request by huang-julien**
### ❓ Type of change
Hi :wave: 

this is a small improvement of server components documentation so users understand that server only components are island components

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/2.guide/2.directory-structure/1.components.md [hunk 1]
```diff
@@ -299,6 +299,10 @@ Now you can register server-only components with the `.server` suffix and use th
 </template>
 ```
 
+Server only components uses `<NuxtIsland>` under the hood, meaning that `lazy` prop and `#fallback` slot are both passed down to `<NuxtIsland>`.
```
Comment (DamianGlowala) [inline docs/2.guide/2.directory-structure/1.components.md:302]:
```suggestion
Server-only components use `<NuxtIsland>` under the hood, meaning that `lazy` prop and `#fallback` slot are both passed down to `<NuxtIsland>`.
```

File: docs/2.guide/2.directory-structure/1.components.md [hunk 2]
```diff
@@ -299,6 +299,10 @@ Now you can register server-only components with the `.server` suffix and use th
...
+
+This also means that server only components are accessible as remote sources and that every features of `<NuxtIsland>` are available on server only components.
```
Comment (DamianGlowala) [inline docs/2.guide/2.directory-structure/1.components.md:304]:
```suggestion
This also means that server-only components are accessible as remote sources and have all features of the `<NuxtIsland>` component available.
```

**Review by DamianGlowala — CHANGES_REQUESTED** [review]

File: docs/2.guide/2.directory-structure/1.components.md [hunk 3]
```diff
@@ -299,6 +299,10 @@ Now you can register server-only components with the `.server` suffix and use th
 </template>
 ```
 
+Server-only components use `<NuxtIsland>` under the hood, meaning that `lazy` prop and `#fallback` slot are both passed down to `<NuxtIsland>`.
+
+This also means that server-only components are accessible as remote sources and have all features of the `<NuxtIsland>` component available.
+
```
Comment (danielroe) [inline docs/2.guide/2.directory-structure/1.components.md:305]: I don't think we currently support specifying sources for server components directly.

```suggestion
```

Comment (huang-julien) [reply docs/2.guide/2.directory-structure/1.components.md:305]: It's technically possible, I think this is a bug.

I also think we should hide server components from being easily accessible by NutxIsland.

https://stackblitz.com/edit/nuxt-starter-pa96dq?file=nuxt.config.ts,components%2FHello.server.vue,components%2FWorld.server.vue,app.vue

There's also a conflict when naming both island and `.server` by the same name

Comment (danielroe) [reply docs/2.guide/2.directory-structure/1.components.md:305]: Oh, I mean the other way around. If you're using a server component - you can't specify the remote source via `source` prop. (Maybe we can reword this sentence instead?)

I think we need NuxtIsland to be able to render arbitrary server components.

Ultimately we'll need 'rules' for rendering islands/server components remotely (that is, which ones to do so, and setting CORS headers for them, where appropriate) and (for example) exposing/exporting component types for remotely rendered components.

But the name conflict sounds like something we should render better errors for.

Comment (huang-julien) [reply docs/2.guide/2.directory-structure/1.components.md:305]: I see :eyes: ...

We can't specify source on `<AsyncServerComponent>`.

So 
- about rules and CORS on server only components. What about a `defineIslandRules` macro ? with `remote` to `true` by default for `island` and `false` for `.server`. I was thinking about hiding `.server` by hashing their names but I don't know if users think that `.server` should be accessible remotely or not
- about the name conflict, a warning log should be enough 🤔 for now ?
- about types, that means we'd need to expose components types in build too ?

Comment (danielroe) [reply docs/2.guide/2.directory-structure/1.components.md:305]: We should discuss types/CORS more but I would disable remote source _consumption_ by default - I would say it's opt-in.

I think a warning log for name should be enough (if there's a conflict), as we currently do when there are multiple components with the same name.

**Review by danielroe — APPROVED** [review]

**Pull request merged**